internal/build/tar.go

```diff
@@ -179,16 +189,18 @@ func (a *ArchiveBuilder) entriesForPath(ctx context.Context, source, dest string
 
 		if sourceIsDir {
 			// Name of file in tar should be relative to source directory...
-			header.Name = strings.TrimPrefix(path, source)
+			tmp, err := filepath.Rel(source, path)
+			if err != nil {
+				return errors.Wrapf(err, "making rel path source:%s path:%s", source, path)
+			}
 			// ...and live inside `dest`
-			header.Name = filepath.Join(dest, header.Name)
-		} else if strings.HasSuffix(dest, string(filepath.Separator)) {
-			header.Name = filepath.Join(dest, filepath.Base(source))
+			header.Name = filepath.Join(dest, tmp)
+		} else if strings.HasSuffix(dest, "/") {
```

Comment:
ya, this should use path.Separator

Reply:
@nicks As i transform everything to forward slash i can't use filepath.Separator (which will be in windows '\' ), everything now can use only one separator that is '/'